### ISSUE TITLE
getaddrinfo EAI_AGAIN graph.microsoft.com

### ISSUE THREAD

**Issue by CHEETX** [ahaenggli/AzureAD-LDAP-wrapper#87]
Hi,

We have AzureAd-LDAP-wrapper 2.0.2 running on Container with DSM 7.2.1-69057 Update 5.
The logs of the container shows a graph error: "getaddrinfo EAI_AGAIN graph.microsoft.com".
This seems to be linked to a problem with the DNS.

The logs:

`graphErrorDetail: [1mnull[22m
2024/07/10 14:02:30	stdout	  error: [32m'getaddrinfo EAI_AGAIN graph.microsoft.com'[39m,
2024/07/10 14:02:30	stdout	  opts: {},
2024/07/10 14:02:30	stdout	  endpoint: [32m"https://graph.microsoft.com/v1.0/users?$count=true&$select=businessPhones,displayName,givenName,jobTitle,mail,mobilePhone,officeLocation,preferredLanguage,surname,userPrincipalName,id,identities,userType,externalUserState,accountEnabled&$filter=userType%20eq%20'Member'%20and%20companyName%20ne%20null%20and%20assignedLicenses%2F%24count%20ne%200"[39m,
2024/07/10 14:02:30	stdout	ERROR: 2024-07-10T12:02:30.011Z:  graph.fetch.js callApi error with {
2024/07/10 14:02:22	stdout	WARN: 2024-07-10T12:02:22.032Z:  graph.fetch.js getGroups() no groups found
2024/07/10 14:02:22	stdout	}`

Any idea to fix this ?

Thanks

Etienne

**Comment by ahaenggli** [comment]
Do you use a proxy or something similar in your network?
If not: maybe it's enough to just set a dns server like this: 
![image](https://github.com/user-attachments/assets/7ea99aa9-626e-423b-b424-bb51fb22fe9a)

**Comment by CHEETX** [comment]
Thank you for the response, I really appreciate your work.
I'm still in the observation phase but I think I've found the solution. Of course, i started by changing the DNS of the Synology by testing with 1.1.1.1 without improvements.

I read on some forums that the node version could influence this error. so I changed "NODE_VERSION" parameters from 18.17.0 to 22.4.1. So far no errors, knock on wood.

Have a good day!
Etienne
